src/pages/MovieInfo.js

```diff
@@ -2,25 +2,50 @@ import Button from 'components/common/Button';
 import { mockArrayData } from 'mock/mockData';
 import './styles.scss';
 import { AiOutlineStar } from 'react-icons/ai';
-import { BsFillBookmarkFill } from 'react-icons/bs';
+import { BsBookmarkPlus } from 'react-icons/bs';
 import { AiFillStar } from 'react-icons/ai';
 import { useHorizontalScroll } from 'hooks/useSideScroll';
 import useTranslation from 'hooks/useTranslation';
-import { useHistory } from 'react-router-dom';
+import { useHistory, useLocation, useParams } from 'react-router-dom';
 import routesPaths from 'routes/routesPaths';
+import { BiArrowBack } from 'react-icons/bi';
+import { setWatchlist, getWatchlist } from 'utils/api';
+import { BsFillBookmarkFill } from 'react-icons/bs';
+import { useEffect, useState } from 'react';
 import Modal from 'components/modalComponent/Modal';
-import { useState } from 'react';
 import { Slider } from 'components/viewComponents/Slider';
 
 export function MovieInfo() {
   const [openModalState, setOpenModalState] = useState(false);
   const [selectedImage, setSelectedImage] = useState('');
   const movie = mockArrayData[0];
-
+  let movieState = useLocation()?.state;
   const history = useHistory();
-
+  let moviesInWatchlist = JSON.parse(getWatchlist());
   const t = useTranslation();
 
+  const findById = id => item => item.id === id;
```

Comment:
You're still duplicating code with this solution. Try this
```suggestion
const findMovieInWatchlist = (id) => (
  moviesInWatchlist.find(movie => movie.id == id)
)
```